proof/crefine/AARCH64/IpcCancel_C.thy

```diff
@@ -1007,269 +878,257 @@ proof -
-               prefer 2
-               apply (frule_tac s=\<sigma> in tcb_queue'_head_end_NULL[symmetric]; simp add: valid_queues_valid_q)
-              apply (rule conjI, solves \<open>clarsimp simp: le_maxDomain_eq_less_numDomains
-                                                        unat_trans_ucast_helper\<close>)
-              apply (frule maxDomain_le_unat_ucast_explicit)
-              apply (clarsimp simp: cready_queues_index_to_C_def numPriorities_def)
-              apply (clarsimp simp: h_val_field_clift' h_t_valid_clift)
-              apply (simp add: t_hrs_ksReadyQueues_upd_absorb)
-
-              apply (rule conjI)
-               apply (clarsimp simp: l2BitmapSize_def' wordRadix_def c_invert_assist)
-
-              apply (subst rf_sr_drop_bitmaps_enqueue_helper, assumption)
-                apply (fastforce intro: cbitmap_L1_relation_bit_set)
-               apply (fastforce intro: cbitmap_L2_relation_bit_set simp: wordRadix_def mask_def)
-              apply (frule_tac d="tcbDomain tcb" and p="tcbPriority tcb" in rf_sr_sched_queue_relation)
-                apply clarsimp
-               apply clarsimp
-              apply (drule_tac qhead'="tcb_ptr_to_ctcb_ptr t" and s=\<sigma> in tcbSchedEnqueue_update,
-                     simp_all add: valid_queues_valid_q)[1]
-               apply (rule tcb_at_not_NULL, erule obj_at'_weakenE, simp)
-              apply (erule(1) state_relation_queue_update_helper[where S="{t}"],
-                     (simp | rule globals.equality)+,
-                     simp_all add: cready_queues_index_to_C_def2 numPriorities_def
-                                   t_hrs_ksReadyQueues_upd_absorb upd_unless_null_def
-                                   typ_heap_simps)[1]
-               apply (fastforce simp: tcb_null_sched_ptrs_def typ_heap_simps c_guard_clift
-                                elim: obj_at'_weaken)+
-             apply (rule conjI, solves \<open>clarsimp simp: le_maxDomain_eq_less_numDomains
-                                                       unat_trans_ucast_helper\<close>)
-             apply clarsimp
-             apply (rule conjI; clarsimp simp: queue_in_range)
-              (* invalid, disagreement between C and Haskell on emptiness of queue *)
-              apply (drule (1) obj_at_cslift_tcb)
-              apply (clarsimp simp: h_val_field_clift' h_t_valid_clift)
-              apply (frule_tac d="tcbDomain tcb" and p="tcbPriority tcb" in rf_sr_sched_queue_relation)
-                apply clarsimp
-               apply clarsimp
-              apply (clarsimp simp: cready_queues_index_to_C_def numPriorities_def)
-              apply (frule_tac s=\<sigma> in tcb_queue'_head_end_NULL)
-               apply (simp add: valid_queues_valid_q)
-              apply clarsimp
-              apply (drule tcb_queue_relation'_empty_ksReadyQueues; simp add: valid_queues_valid_q)
-             (* queue was not empty, add t to queue and leave bitmaps alone *)
```

Comment:
I have minor concerns that these informative comments went away in the new proof, and no new comments replaced them.

Reply:
The proofs in CRefine are now super boring, as I intended all along. There are the `ccorres` proofs for the library functions, and then for `ccorres` proofs `tcbSchedEnqueue/Append/Dequeue` are little more than using these new rules. Unfortunately these `ccorres` proofs are still quite verbose, mainly because there are so many `threadSet` operations, each of which is handled with a long-winded `ccorres` step. 

The difficult parts of the `ccorres` proofs that warranted these informative comments are now the difficult parts of the `corres` proofs in Refine, where I do have comments similar to this (though due to the separation of the bitmap rules, there's no mixing any more)

Reply:
That should be fine. Thanks.